runbotics/runbotics-desktop/src/actions/microsoft/microsoft.service.ts

```diff
@@ -68,16 +69,35 @@ export class MicrosoftService {
         return downloadLink;
     }
 
-    async uploadFile(token: string, sharepointFileName: string, contentType, content, cloudPath) {
+    async getDownloadFileLinkFromSite(token: string, fileName: string, folderPath?: string, ) {
+        this.TOKEN = token;
+        let url;
+        if (folderPath) {
+            url = `https://graph.microsoft.com/v1.0/sites/${this.siteId}/drives/${this.driveId}/root:/${folderPath}:/children`;
+        } else {
+            url = `https://graph.microsoft.com/v1.0/sites/${this.siteId}/drives/${this.driveId}/root/children`;
+        }
+
+        const authHeaders = this.getAuthHeader();
+        const { data } = await MicrosoftService.makeRequest().get<APICell>(url, {
+            headers: {
+                ...authHeaders,
+            },
+        });
+        const downloadLink = data.value.find((item) => item.name === fileName)?.['@microsoft.graph.downloadUrl'];
+        return downloadLink;
+    }
+
+    async uploadFile(token: string, filePath: string, cloudPath: string, contentType: string, content) {
         this.TOKEN = token;
         let url = '';
 
         switch (cloudPath) {
-            case 'root':
-                url = `https://graph.microsoft.com/v1.0/me/drive/root:/${sharepointFileName}:/content`;
+            case CloudPath.Root:
```

Comment:
2 variables have the same name with minor diff, rename one of them, try to stick to camel case